### PR TITLE
feat: add channel flags to all channels

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

adds channel flags to all currently supported channels.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/threads.py [hunk 1]
```diff
@@ -332,6 +332,14 @@ def created_at(self) -> datetime.datetime:
         """
         return self.create_timestamp or snowflake_time(self.id)
 
+    @property
+    def flags(self) -> ChannelFlags:
+        """:class:`ChannelFlags`: The channel flags for this channel.
+
+        .. versionadded:: 2.6
```
Comment (shiftinv) [inline disnake/threads.py:339]:
```suggestion
        .. versionadded:: 2.5
```
also duplicated documentation in class docstring, that one should be removed

File: disnake/types/channel.py [hunk 2]
```diff
@@ -53,6 +53,7 @@ class _BaseGuildChannel(_BaseChannel):
     permission_overwrites: List[PermissionOverwrite]
     nsfw: bool
     parent_id: Optional[Snowflake]
+    flags: int
```
Comment (shiftinv) [inline disnake/types/channel.py:56]: optional according to docs, though pretty sure I haven't seen that so far

Comment (onerandomusername) [reply disnake/types/channel.py:56]: Its treated as optional everywhere, might as well change this to mark it as optional.

Comment (onerandomusername) [reply disnake/types/channel.py]: https://github.com/DisnakeDev/disnake/pull/547/commits/e8e04ad9074981e00f332ecd131afd9cca008449

File: disnake/threads.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/threads.py]: https://github.com/DisnakeDev/disnake/pull/547/commits/73adfc9a7d5d67329151450f21b3288079efad4b

File: disnake/abc.py [hunk 3]
```diff
@@ -540,6 +541,14 @@ def permissions_synced(self) -> bool:
         category = self.guild.get_channel(self.category_id)
         return bool(category and category.overwrites == self.overwrites)
 
+    @property
+    def flags(self) -> ChannelFlags:
+        """:class:`ChannelFlags`: The channel flags for this channel.
```
Comment (shiftinv) [inline disnake/abc.py:546]:
```suggestion
        """:class:`.ChannelFlags`: The channel flags for this channel.
```

Comment (onerandomusername) [reply disnake/abc.py]: https://github.com/DisnakeDev/disnake/pull/547/commits/49d1fdc15b62d8cc9b04e53d27c4cdc25a425cb9

File: disnake/channel.py [hunk 4]
```diff
@@ -2813,6 +2820,7 @@ def __init__(self, *, me: ClientUser, state: ConnectionState, data: DMChannelPay
         self.last_pin_timestamp: Optional[datetime.datetime] = utils.parse_time(
             data.get("last_pin_timestamp")
         )
+        self._flags: int = data.get("flags", 0)
```
Comment (shiftinv) [inline disnake/channel.py:2823]: missing in `.types.channel.DMChannel`

**Review by shiftinv — APPROVED** [review]
lgtm

**Pull request merged**